src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -65,11 +65,12 @@ public sealed partial class JsonSerializerOptions
         /// </summary>
         public JsonSerializerOptions()
         {
-            _converters = new ConfigurationList<JsonConverter>(this);
+            _converters = new ConfigurationList<JsonConverter>() { VerifyMutable = VerifyMutable };
```

Comment:
This is allocating an additional delegate. You might want to consider the following pattern:
```C#
new ConfigurationList<JsonConverter> { State = this ; VerifyMutable = static state=> ((JsonSerializerOptions)state!).VerifyMutable(); }
```
alternatively we could have a `ConverterList` inherit from `ConfigurationList<JsonConverter>` and pass the options instance directly as a constructor parameter.

Reply:
isn't this microoptimization? in normal usage scenario that's a single allocation per app?